test/gtest/ucs/test_async.cc

```diff
@@ -349,10 +375,46 @@ class test_async_mt : public test_async {
         return result;
     }
 
-    void spawn() {
+    int repeat_create_destroy_run(unsigned index) {
+        static_assert(std::is_base_of<local_event, LOCAL>::value, "");
+        LOCAL *le;
+        m_ev[index] = le = new LOCAL(GetParam());
+
+        check_is_blocked(le, false);
+
+        barrier();
+
+        while (!m_stop[index]) {
+            le->block();
+            check_is_blocked(le, true);
+            unsigned before = le->count();
+            suspend_and_poll(le, 10);
+            unsigned after = le->count();
+            le->unblock();
+            EXPECT_EQ(before, after); /* Should not handle while blocked */
+
+            auto wait = ucs::rand() % 20;
+            le->unset_event();
+            le->destory_event();
+            suspend(wait);
+            le->create_event(); /* same memory could be used by other thread */
+            le->set_event();
+            suspend(20 - wait);
+            le->check_miss();
+        }
+
+        check_is_blocked(le, false);
+
+        int result = le->count();
+        delete le;
```

Comment:
pls consider using std::scoped_ptr to get rid of explicit delete
You can also just `return le->count();` in this case without saving in tmp var